package.json

```diff
@@ -62,15 +62,32 @@
     ],
     "configuration": {
       "type": "object",
-      "title": "Slice",
+      "title": "Slice Settings",
       "properties": {
-        "slice.referenceDirectories": {
+        "slice.configurations": {
           "type": "array",
+          "title": "Slice Configurations",
+          "description": "Array of Slice configuration sets.",
           "items": {
-            "type": "string"
-          },
-          "description": "The directories containing the target Slice reference files.",
-          "scope": "resource"
+            "type": "object",
+            "properties": {
+              "referenceDirectories": {
+                "type": "array",
+                "items": {
+                  "type": "string"
+                },
+                "description": "The directories containing the target Slice reference files."
+              },
+              "includeBuiltInTypes": {
+                "type": "boolean",
+                "default": true,
+                "description": "Enables / Disables the built-in Slice types."
+              }
+            },
+            "required": [
```

Comment:
It means if you add settings like this 

```
"slice.configuration" : [
{
}
]
```

It will yell at you to add reference directories

```
"slice.configuration" : [
{
  "referenceDirectories": ["foo"]
}
]
```